File: qt/network/gen_qhttp1configuration.go
package network

import (
	"unsafe"
)

type QHttp1Configuration struct {
	h          uintptr
	isSubclass bool
}

// NewQHttp1Configuration constructs a new QHttp1Configuration object.
func NewQHttp1Configuration() *QHttp1Configuration {

	ret := newQHttp1Configuration(QHttp1Configuration_new())
	ret.isSubclass = true
	return ret
}

// NewQHttp1Configuration2 constructs a new QHttp1Configuration object.
func NewQHttp1Configuration2(other *QHttp1Configuration) *QHttp1Configuration {

	ret := newQHttp1Configuration(QHttp1Configuration_new2(other.cPointer()))
	ret.isSubclass = true
	return ret
}

func (this *QHttp1Configuration) OperatorAssign(other *QHttp1Configuration) {
	QHttp1Configuration_OperatorAssign(this.h, other.cPointer())
}

func (this *QHttp1Configuration) SetNumberOfConnectionsPerHost(amount int64) {
	QHttp1Configuration_SetNumberOfConnectionsPerHost(this.h, (ptrdiff_t)(amount))
}

func (this *QHttp1Configuration) NumberOfConnectionsPerHost() int64 {
	return (int64)(QHttp1Configuration_NumberOfConnectionsPerHost(this.h))
}

func (this *QHttp1Configuration) Swap(other *QHttp1Configuration) {
	QHttp1Configuration_Swap(this.h, other.cPointer())
}

File: qt/network/gen_qhttp1configuration.cpp
// +build ignore

#include <QHttp1Configuration>
#include <qhttp1configuration.h>
#include "gen_qhttp1configuration.h"

QHttp1Configuration* QHttp1Configuration_new() {
	return new QHttp1Configuration();
}

QHttp1Configuration* QHttp1Configuration_new2(QHttp1Configuration* other) {
	return new QHttp1Configuration(*other);
}

void QHttp1Configuration_OperatorAssign(QHttp1Configuration* self, QHttp1Configuration* other) {
	self->operator=(*other);
}

void QHttp1Configuration_SetNumberOfConnectionsPerHost(QHttp1Configuration* self, ptrdiff_t amount) {
	self->setNumberOfConnectionsPerHost((qsizetype)(amount));
}

ptrdiff_t QHttp1Configuration_NumberOfConnectionsPerHost(const QHttp1Configuration* self) {
	qsizetype _ret = self->numberOfConnectionsPerHost();
	return static_cast<ptrdiff_t>(_ret);
}

void QHttp1Configuration_Swap(QHttp1Configuration* self, QHttp1Configuration* other) {
	self->swap(*other);
}

void QHttp1Configuration_Delete(QHttp1Configuration* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QHttp1Configuration*>( self );
	} else {
		delete self;
	}
}

